runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1224,20 +1715,125 @@ TR::Register *J9::Power::TreeEvaluator::awrtbariEvaluator(TR::Node *node, TR::Co
    return NULL;
    }
 
+TR::Register *J9::Power::TreeEvaluator::fwrtbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   // For rdbar and wrtbar nodes we first evaluate the children we need to
+   // handle the side effects. Then we delegate the evaluation of the remaining
+   // children and the load/store operation to the appropriate load/store evaluator.
+   TR::Node *sideEffectNode = node->getSecondChild();
+   TR::Register *valueReg = cg->evaluate(node->getFirstChild());
+   TR::Register *sideEffectRegister = cg->evaluate(sideEffectNode);
+   if (cg->comp()->getOption(TR_EnableFieldWatch))
+      {
+      TR::TreeEvaluator::rdWrtbarHelperForFieldWatch(node, cg, sideEffectRegister, valueReg);
+      }
+   // The Value Node, or the second child is not decremented here. The store evaluator also uses it, and decrements it.
```

Comment:
Should be `or the first child ` instead. Similarly everywhere else in the static/direct opcode evaluators.

Reply:
Fixed here:
https://github.com/eclipse/openj9/pull/6234/files#diff-efcdf3eda93e6ae70d4c46dd0bf9b5acR1929

@dchopra001 going to resolve this one on your behalf.